### ISSUE TITLE
Connection shows "waiting" forever even after a connection was successful

### ISSUE THREAD

**Issue by adamstallard** [BrightID/BrightID#387]
Jon tried to connect for the first time to someone.  He tried several times, and each time it showed "waiting" and that text stayed on the connection list next to the new connection forever.

The connection was in fact successful at least one of the times.  The evidence is that it says "Last connected 22 minutes ago."

![09D4E1DB-64FE-493D-A354-79179D78FEE1](https://user-images.githubusercontent.com/205792/79807618-5535c200-8320-11ea-8a56-f1046ef41c3a.png)

**Comment by x5engine** [comment]
How can reproduce this?

**Comment by x5engine** [comment]
Got the bug even if I have other connections but the other account had no connections

![image](https://user-images.githubusercontent.com/12363020/79808797-68678280-835e-11ea-99d8-f7311408a1b5.png)

**Comment by x5engine** [comment]
after a few minutes now it's connected. so how can I remove a connection to re-test this again?

**Comment by x5engine** [comment]
on my phone, it's still pending 
![image](https://user-images.githubusercontent.com/12363020/79811907-d2842580-8366-11ea-8079-36cb3c1a332a.png)

**Comment by TripleSpeeder** [comment]
@x5engine can you still reproduce this?

**Comment by x5engine** [comment]
yes I think so, @TripleSpeeder just try a new device and add a connection to another account which already had some connections... we can try to debug it together

**Comment by adamstallard** [comment]
We will recheck this bug when reusable QR codes have been implemented #419

**Comment by adamstallard** [comment]
There is a related issue #364 to change the message from "waiting" to "failed" if it's clear it's not going to connect.

**Comment by TripleSpeeder** [comment]
Copying my investigation results here so this does not get buried in the chat. Like Adam posted above, this issue should be fixed when we have reusable QR-Codes (#419).

<hr>
Assume UserA, UserB and UserC exist. UserA wants to connect with both UserB and UserC, so he pulls up his QR-Code and shows it to both UserB and UserC. 

 1. While UserA is in the "My Code" screen, the app will continue to poll for uploaded profiles of the connection partner.
 1. UserB scans the code and sees the Connection Preview screen for UserA.
 1. UserC scans the code and also sees the Connection Preview screen for UserA.
 1. UserB hits the "Confirm" button and gets the "Sucess" message. In the background this triggers upload of UserB's profile and signed connection message. 
 1. UserA gets the the profile and signed connection message from UserB. He executes the addCOnnection operation and gets the sucess message. UserA's app will stop looking for profile uploads as soon as the "My Code" screen is left.
 1. Shortly after UserB also UserC clicks the "confirm" button. Immediately he gets the "Success" message and in the background his profile and signed connection message is uploaded.
 1. But UserA is not waiting anymore for a profile upload. UserA will never see a Connection Preview screen for UserC.

Result:
UserC has the connection to UserA only locally recorded. Therefor It is in "Waiting" state forever, as UserA has no way to confirm the connection. UserA does not even know that UserC has scanned his code and is waiting!

There is another path leading to the same result even when only 2 people are involved:
1. UserB scans the code of UserA and sees the Connection Preview screen for UserA.
2. UserB does not immediately click confirm
3. UserA wants to quickly check his groups, so he hits the "Groups" button.
   -> UserA's app will stop polling for profileuploads as soon as the "My Code" screen is left
4. Rest as above - When UserB clicks confirm and uploads his profile, UserA has already stopped looking for it. So UserB has UserA added only locally and is stuck in "waiting" forever.
